lib/web_ui/dev/test_runner.dart

```diff
@@ -279,8 +308,42 @@ class TestCommand extends Command<bool> with ArgUtils {
   bool get isFirefox => browser == 'firefox';
 
   /// Whether [browser] is set to "safari".
-  bool get isSafariOnMacOS => browser == 'safari'
-      && io.Platform.isMacOS;
+  bool get isSafariOnMacOS => browser == 'safari' && io.Platform.isMacOS;
+
+  /// Due to efficiancy constraints, Chrome integration tests only run on
+  /// Linux on LUCI.
+  ///
+  /// They run on all platforms for local.
+  bool get isChromeIntegrationTestAvailable =>
+      (isChrome && isLuci && io.Platform.isLinux) || (isChrome && !isLuci);
+
+  /// Due to efficiancy constraints, Firefox integration tests only run on
+  /// Linux on LUCI.
+  ///
+  /// For now Firefox integration tests only run on Linux and Mac on local.
+  ///
+  // TODO: https://github.com/flutter/flutter/issues/63832
+  bool get isFirefoxIntegrationTestAvailable =>
+      (isFirefox && isLuci && io.Platform.isLinux) ||
+      (isFirefox && !isLuci && !io.Platform.isWindows);
+
+  /// Latest versions of Safari Desktop are only available on MacOS.
+  ///
+  /// Integration testing on LUCI is not supported at the moment.
+  // TODO: https://github.com/flutter/flutter/issues/63710
+  bool get isSafariIntegrationTestAvailable =>
+      (isSafariOnMacOS && isLuci && io.Platform.isLinux) ||
+      (isSafariOnMacOS && !isLuci && !io.Platform.isWindows);
```

Comment:
Checking `!io.Platform.isWindows` is redundant as `isSafariOnMacOS` getter already checks for `io.Platform.isMacOS`.

Reply:
this is deleted.